frappe/tests/test_naming.py

```diff
@@ -51,6 +51,40 @@ def test_field_autoname_name_sync(self):
 		self.assertEqual(country.name, original_name)
 		self.assertEqual(country.name, country.country_name)
 
+	def test_child_table_naming(self):
+		child_dt_with_naming = new_doctype(
+			"childtable_with_autonaming", istable=1, autoname="field:some_fieldname"
+		).insert()
+		dt_with_child_autoname = new_doctype(
+			"dt_with_childtable_naming",
+			fields=[
+				{
+					"label": "table with naming",
+					"fieldname": "table_with_naming",
+					"options": "childtable_with_autonaming",
+					"fieldtype": "Table",
+				}
+			],
+		).insert()
+
+		name = frappe.generate_hash(length=10)
+
+		doc = frappe.new_doc("dt_with_childtable_naming")
+		doc.append("table_with_naming", {"some_fieldname": name})
+		doc.save()
+		self.assertEqual(doc.table_with_naming[0].name, name)
+
+		# change autoname field
+		doc.table_with_naming[0].some_fieldname = "Something else"
+		doc.save()
+
+		self.assertEqual(doc.table_with_naming[0].name, name)
+		self.assertEqual(doc.table_with_naming[0].some_fieldname, name)
+
+		doc.delete()
+		dt_with_child_autoname.delete()
+		child_dt_with_naming.delete()
```

Comment:
This should be handled by testcase :P 

the queue of docs to be discarded after test regardless of test status.